File: 290/py/290.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        n = len(pattern)

        arrS = s.split(" ")
        m = len(arrS)

        if n != m :
            return False

        # make dict
        mapS = {}
        mapP = {}

        for i in range(0, n):
            st = arrS[i]
            p = pattern[i]

            if mapS.get(st) :
                if mapS.get(st) != p:
                    return False
            else:
                mapS[st] = p
            
            if mapP.get(p) :
                if mapP.get(p) != st:
                    return False
            else:
                mapP[p] = st
        
        return True

File: 290/go/290.go
func wordPattern(pattern string, s string) bool {
    n := len(pattern)
    // turn s into slice
    sliceS := strings.Split(s, " ")
    m := len(sliceS)

    if n != m {
        return false
    }

    // make double map
    mapP := make(map[byte]string)
    mapS := make(map[string]byte)

    for i:= 0; i < n; i++ {
        str := sliceS[i]
        byt := pattern[i]

        valS, findS := mapS[str]
        if findS {
            if valS != byt {
                return false
            }
        }else{
            mapS[str] = byt
        }

        valP, findP := mapP[byt]
        if findP {
            if valP != str {
                return false
            }
        }else{
            mapP[byt] = str
        }

    }

    return true
}